libs/dataRequests.tsx

```diff
@@ -1,17 +1,50 @@
 "use server";
+import moment from "moment";
 import { redirect } from "next/navigation";
 
 const USERNAME = process.env.NEXT_PRIVATE_USERNAME;
 const PASSWORD = process.env.NEXT_PRIVATE_PASSWORD;
 const BACKEND_URL = process.env.NEXT_PUBLIC_BACKEND_URL;
 
+const backendUrlSplit = BACKEND_URL?.split("://");
+const scheme = backendUrlSplit?.[0];
+const domain = backendUrlSplit?.[1];
+
+const logInfo = (message: string, method: string, url: string) => {
+  console.info({
+    event: message,
+    http: {
+      method: method,
+      scheme: scheme,
+      host: domain,
+      // port: // TODO add port number
+      path: url,
+      started_at: moment(new Date()).format("YYYY-MM-DDTHH:mm:ss.SSSZZ"),
+    },
+  });
+};
+
+const logError = (message: string, method: string, url: string) => {
+  console.error({
```

Comment:
is there any version of a stack trace in frontend land or would that be what we're passing in as the message?

if its in the message I'd probably use that to populate the error field and use a simple english message for event, i.e

```
message: "Rhys saying whats gone wrong where and when in an idiot friendly way, i.e failed to get x`
error: [whatever error we're getting back - the more raw message from the system for whoever needs to figure out the issue]
```

does that make sense?

Reply:
Swtiched up how tthe errors are handled, so we can grab the stack. Following the dp standards, so it'll be `errors: [{our error}}` will output tto something similar as below.
![image](https://github.com/GSS-Cogs/idpd-frontend-datasets/assets/110108574/51526692-0822-4bf0-89d7-8116959e456f)